components/empty-state/empty-state-illustrated-link.js

```diff
@@ -22,74 +18,17 @@ class EmptyStateIllustratedLink extends RtlMixin(LitElement) {
 			  * @type {string}
 			  */
 			actionHref: { type: String, attribute: 'action-href' },
-			/**
-			 * Optional: The action text to be used in the link
-			 * @type {string}
-			 */
-			actionText: { type: String, attribute: 'action-text' },
-			/**
-			 * REQUIRED: A description giving details about the empty state
-			 * @type {string}
-			 */
-			description: { type: String },
-			/**
-			 * Optional: The name of the preset image you would like to display in the component
-			 * @type {string}
-			 */
-			illustrationName: { type: String, attribute: 'illustration-name' },
-			/**
-			 * REQUIRED: A title for the empty state
-			 * @type {string}
-			 */
-			titleText: { type: String, attribute: 'title-text' },
-			_contentWidth: { type: Number }
 		};
 	}
 
-	static get styles() {
-		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
-	}
-
 	constructor() {
 		super();
-
-		this._contentWidth = 700;
-		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
-	}
-
-	connectedCallback() {
-		super.connectedCallback();
-		this._resizeObserver.observe(this);
-	}
-
-	disconnectedCallback() {
-		super.disconnectedCallback();
-		this._resizeObserver.disconnect();
-	}
-
-	firstUpdated(changedProperties) {
-		super.firstUpdated(changedProperties);
-
-		if (!this.titleText || this.titleText.length === 0) {
-			console.warn('d2l-empty-state-illustrated-link component requires titleText.');
-		}
-
-		if (!this.description || this.description.length === 0) {
-			console.warn('d2l-empty-state-illustrated-link component requires a description.');
-		}
+		this._illustratedComponentType = 'link';
 	}
 
 	render() {
-		const illustrationContainerStyle = {
-			height: `${Math.min(this._contentWidth, 500) / 1.5}px`,
-		};
-
-		const titleSmall = this._contentWidth <= 615;
-		const titleClass = {
-			'd2l-empty-state-title': true,
-			'd2l-empty-state-title-small': titleSmall,
-			'd2l-empty-state-title-large': !titleSmall,
-		};
+		const illustrationContainerStyle = this._getIllustrationContainerStyle();
+		const titleClass = this._getTitleClass();
 
 		const actionLink = this.actionText && this.actionHref
 			? html`<d2l-link class='d2l-body-compact d2l-empty-state-action' href=${this.actionHref}>${this.actionText}</d2l-link>`
```

Comment:
A good general practice we've been using to try to cut down on the number of web components on a page (since they each cost a little performance) is to use a regular `<a>` whenever we can inside our Lit components.

So you could import `linkStyles` from `../link/link.js`, add those to the styles array and add `d2l-link` to the CSS classes on this and switch it to a `<a>`.

Reply:
Oh also, double quotes around the attributes for consistency with other stuff.